implementations/rust/ockam/ockam_channel/src/secure_channel.rs

```diff
@@ -52,24 +52,25 @@ impl SecureChannel {
         A: Into<Address>,
         N: SecureChannelNewKeyExchanger,
         V: SecureChannelVault,
+        C: NodeContext,
     >(
-        ctx: &Context,
+        ctx: &C,
         address: A,
         new_key_exchanger: N,
         vault: V,
     ) -> Result<()> {
         let address = address.into();
         let channel_listener = SecureChannelListener::new(new_key_exchanger, vault);
         info!("Starting SecureChannel listener at {}", &address);
-        ctx.start_worker(address, channel_listener).await?;
+        ctx.start_worker(address.into(), channel_listener).await?;
```

Comment:
Is this a new necessity of the `Context` trait (i.e. will users have to call `.into()` manually on types that they pass to the new context API?

Reply:
The issue is that you end up needing `Into<Route> + Send + 'static`, or the resulting future is fairly restricted due to the way `async_trait` desugars. This ends up being pretty restrictive and leading to confusing lifetime errors for users, but perhaps it's worth it?

Note that we still have the inherent methods, so it doesn't impact the examples, mostly the internals. This also might be alleviated with proper (non-`dyn`) async trait support.